pandas/core/index.py

```diff
@@ -1181,15 +1181,23 @@ def argsort(self, *args, **kwargs):
     def __add__(self, other):
         if isinstance(other, Index):
             warnings.warn("using '+' to provide set union with Indexes is deprecated, "
-                          "use '|' or .union()",FutureWarning)
+                          "use '|' or .union()", FutureWarning)
             return self.union(other)
         return Index(np.array(self) + other)
+
+    def __radd__(self, other):
+        if isinstance(other, Index):
```

Comment:
This code path will never be called, because `a.__radd__(b)` is only called when `b.__add__(a)` returns `NotImplemented` -- which, as you can see above, never happens if both objects are `Index` instances.